Production_scripts/B05a_Process_A&E_(AE2)_extract.R

```diff
@@ -0,0 +1,950 @@
+    lca = "Council Area Code",
+    ae_disdest = "Discharge Destination Code",
+    keyTime2 = "DAT Time",
+    diag1 = "Disease 1 Code",
+    diag2 = "Disease 2 Code",
+    diag3 = "Disease 3 Code",
+    falls_adm = "Falls Related Admission",
+    gpprac = "GP Practice Code",
+    hscp = "HSCP of Residence Code - current",
+    hbrescode = "NHS Board of Residence Code - current",
+    hbtreatcode = "Treatment NHS Board Code - current",
+    chi = "Pat UPI [C]",
+    gender = "Pat Gender Code",
+    ae_patflow = "Patient Flow Code",
+    ae_placeinc = "Place of Incident Code",
+    ae_reasonwait = "Reason for Wait Code",
+    refsource = "Referral Source Code",
+    selfharm_adm = "Self Harm Related Admission",
+    submis_adm = "Substance Misuse Related Admission",
+    sigfac = "Significant Facility Code",
+    cost_total_net = "Total Net Costs",
+    location = "Treatment Location Code",
+    case_ref_number = "Case Reference Number"
+  )
+
+# Data Cleaning -----------------------------------------
+
+# year variable
+ae_clean <- ae_file %>%
+  mutate(
+    year = year,
+    recid = "AE2"
+  ) %>%
+  ## Recode GP Practice into a 5 digit number ##
+  # assume that if it starts with a letter it's an English practice and so recode to 99995
+  convert_eng_gpprac_to_dummy(gpprac) %>%
+  # use the CHI postcode and if that is blank, then use the epi postcode.
+  mutate(postcode = if_else(!is.na(postcode_chi), postcode_chi, postcode_epi)) %>%
+  ## recode cypher HB codes ##
+  mutate(across(c(hbtreatcode, hbrescode), ~ case_when(
+    .x == "A" ~ "S08000015",
+    .x == "B" ~ "S08000016",
+    .x == "F" ~ "S08000029",
+    .x == "G" ~ "S08000021",
+    .x == "H" ~ "S08000022",
+    .x == "L" ~ "S08000023",
+    .x == "N" ~ "S08000020",
+    .x == "R" ~ "S08000025",
+    .x == "S" ~ "S08000024",
+    .x == "T" ~ "S08000030",
+    .x == "V" ~ "S08000019",
+    .x == "W" ~ "S08000028",
+    .x == "Y" ~ "S08000017",
+    .x == "Z" ~ "S08000026"
+  ))) %>%
+  ## Allocate the costs to the correct month ##
+  # Create month variable
+  mutate(month = strftime(record_keydate1, "%m")) %>%
+  # Allocated cost in correct month
+  mutate(
+    apr_cost = if_else(month == "04", cost_total_net, 0),
+    may_cost = if_else(month == "05", cost_total_net, 0),
+    jun_cost = if_else(month == "06", cost_total_net, 0),
+    jul_cost = if_else(month == "07", cost_total_net, 0),
+    aug_cost = if_else(month == "08", cost_total_net, 0),
+    sep_cost = if_else(month == "09", cost_total_net, 0),
+    oct_cost = if_else(month == "10", cost_total_net, 0),
+    nov_cost = if_else(month == "11", cost_total_net, 0),
+    dec_cost = if_else(month == "12", cost_total_net, 0),
+    jan_cost = if_else(month == "01", cost_total_net, 0),
+    feb_cost = if_else(month == "02", cost_total_net, 0),
+    mar_cost = if_else(month == "03", cost_total_net, 0)
+  ) %>%
+  # sort for linking on CUP marker
+  arrange(record_keydate1, keyTime1, case_ref_number)
```

Comment:
Don't need all these sorts in R. They're only required in SPSS for matching files together.